packages/smooth_app/lib/database/tmp.dart

```diff
@@ -0,0 +1,112 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:http/http.dart';
+import 'package:openfoodfacts/model/KnowledgePanels.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:openfoodfacts/utils/QueryType.dart';
+import 'package:openfoodfacts/utils/UriHelper.dart';
+
+/// Fixes to off-dart.
+// TODO(monsieurtanuki): to be moved to off-dart
+class OpenFoodAPIClientTmp {
```

Comment:
So I guess you created this in order to return string from apis and store strings in the `_localDatabase`?

Tbh I'm not sure all this work is worth it, it also feels a bit brittle. We are replicating logic from the dart package (how to parse the String -> Product or String -> KnowledgePanel), but this code belongs to the dart package, if it changes there it will get outdated here. Also just feels a little off, after all this is the dart package's responsibility... No?

Also looks like it would be fairly easy to live without all of this:

`OnboardingDataProduct` and `OnboardingDataKnowledgePanels` define their own `getData` and `downloadData` methods:

```
class OnboardingDataProduct {
 KnowledgePanels _panels;
 KnowledgePanels getData() {
   return _panels;
}

  void downloadData() async {
   // call OpenFoodFacts-dart to fetch KnowledgePanels
  _panels = await openfoodfactsdart.getKnowledgePanels();
}
}
```

Same thing for the product.

Reply:
That's correct, that's the off-dart package responsibility to convert to/from strings. That's why I named that file `tmp.dart` and added a `TODO` about moving the code to off-dart - I needed to check first if it worked and if it passed the review in smoothie :)

I believe it's the perfect use case where we see that off-dart needs improvements: like most people, I don't know how to store objects, but I'm good at storing strings. And if off-dart directly gives me an object I have to code an ugly toJson().toString() part if I want to store locally an object.
But if I do it in several steps: 1. retrieve a string from the back-end 2. convert that string to an object, I can also 3. store locally the string and 4. retrieve again that string and check if it's the same (instead of always creating an object and refreshing the display with identical data).

There are other cases where it makes sense to split download and conversion.
For the nutrients we need to download the localized list (that is probably not going to change very often). If we store the string and then download and just compare the strings, we're able to say "ok nothing's changed, let's not refresh the display".
Same for the localized "importances" and preferences: at EVERY init we download them and REFRESH the display, regardless of any change. And I don't like that useless display stuttering - as an end-user.

Reply:
I think the dart package is an abstraction on top so that the clients don't have to worry about strings.

If you do have to store strings, the best thing to do is to add `serialize` (or override `toString`) and `deserialize` methods on the objects `KnowledgePanel` and `Product`. I would also recommend adding a `equals` or `compare` methods if needed rather than letting the client compare strings.

Reply:
> I think the dart package is an abstraction on top so that the clients don't have to worry about strings.

Sure, most clients won't have to worry about strings - so far we did not. But the clients who want to save data locally will. And those who are interested in performance too.

> If you do have to store strings, the best thing to do is to add `serialize` (or override `toString`) and `deserialize` methods on the objects `KnowledgePanel` and `Product`.

I think we already have more or less those methods, with `toJson` and `fromJson`.

> I would also recommend adding a `equals` or `compare` methods if needed rather than letting the client compare strings.

If you want to code and maintain `equals` for `Product`, be my guest.

Reply:
if needed, would be happy to add `equals` :) 

But here, I'm really arguing whether it's realistic for off-dart to return String instead of the object (so more specifically if this tmp.dart is needed). If the client has to store Strings I don't think it's too bad for them to call `toJson().toString()` [or we can override `toString` on the objects for convenience that does `toJson().toString()`]. If the client wants to play with strings it should be their choice, but the dart package should always give objects back. Just my 2c :) 

Approving PR, I think I gave my opinion, feel free to make your call :)